src/IHaskell/Eval/Evaluate.hs

```diff
@@ -1093,10 +1052,10 @@ keepingItVariable act = do
   goStmt $ printf "let it = %s" itVariable
   act
 
-capturedStatement :: (String -> IO ())         -- ^ Function used to publish intermediate output.
-                  -> String                            -- ^ Statement to evaluate.
-                  -> Interpreter (String, RunResult)   -- ^ Return the output and result.
-capturedStatement output stmt = do
+capturedEval :: (String -> IO ()) -- ^ Function used to publish intermediate output.
+             -> Either String (IO a) -- ^ Statement to evaluate.
```

Comment:
Replace `Either` with a custom data type:
```haskell
data CapturedEval a = CapturedStmt String | CapturedIO (IO a)
```
Otherwise this looks like the classic `Either` with an error message `String` attached to it.